File: package/main/src/Object/isEmpty.ts
/**
 * Checks if an object is empty (has no own properties)
 * @param object - The object to check
 * @returns true if the object is empty, false otherwise
 */
export const isEmpty = (object: Record<string, unknown>): boolean => {
  // Check enumerable string keys
  for (const key in object) {
    if (Object.hasOwn(object, key)) {
      return false;
    }
  }

  // Check symbol properties
  const symbolKeys = Object.getOwnPropertySymbols(object);
  if (symbolKeys.length > 0) {
    return false;
  }

  return true;
};

File: package/umt_rust/src/object/is_empty.rs
use super::Value;
use std::collections::HashMap;

/// Checks if an object is empty (has no own properties).
///
/// # Arguments
///
/// * `object` - The object to check
///
/// # Returns
///
/// Returns true if the object is empty, false otherwise.
///
/// # Examples
///
/// ```
/// use umt_rust::object::{umt_is_empty, Value};
/// use std::collections::HashMap;
///
/// let obj = HashMap::new();
/// assert!(umt_is_empty(&obj));
/// ```
pub fn umt_is_empty(object: &HashMap<String, Value>) -> bool {
    object.is_empty()
}
